api/storage_service.cc

```diff
@@ -1403,16 +1410,23 @@ void set_snapshot(http_context& ctx, routes& r, sharded<db::snapshot_ctl>& snap_
         } else {
             throw httpd::bad_param_exception(fmt::format("Unknown argument for 'quarantine_mode' parameter: {}", quarantine_mode_str));
         }
+
         return f.then([&ctx, keyspace, column_families, opts] {
-            return ctx.db.invoke_on_all([=] (replica::database& db) {
-                return do_for_each(column_families, [=, &db](sstring cfname) {
+            return ctx.db.map_reduce0([=] (replica::database& db) {
+                return map_reduce(column_families, [=, &db] (sstring cfname) {
                     auto& cm = db.get_compaction_manager();
                     auto& cf = db.find_column_family(keyspace, cfname);
                     return cm.perform_sstable_scrub(cf.as_table_state(), opts);
-                });
-            });
-        }).then([]{
-            return make_ready_future<json::json_return_type>(0);
+                }, compaction_manager::task::compaction_task_stats{}, std::plus<compaction_manager::task::compaction_task_stats>());
+            }, compaction_manager::task::compaction_task_stats{}, std::plus<compaction_manager::task::compaction_task_stats>());
+        }).then_wrapped([] (auto f) {
+            if (f.failed()) {
+                return make_ready_future<json::json_return_type>(static_cast<int>(scrub_return::aborted));
+            } else if (f.get().validation_errors == 0) {
```

Comment:
`f.get0()` is more appropriate here.

Reply:
`get0()` is deprecated, we no longer support variadic futures.

Reply:
is get() ok then?